### ISSUE TITLE
Specific occupation roles and occupation dispositions are needed for each occupation

### ISSUE THREAD

**Issue by mattolds** [Occupation-Ontology/OccO#16]
**Describe the bug**
I read your FOIS 2023 paper. If I'm misinterpreting, I apologize, but according to section 2.2, which lays out the OccO design plan (screenshot below), chunks of the ontology are currently missing—specifically, a specific occupation role and occupation disposition for each occupation. As is, the OccO only captures the bearers of specific occupations (i.e., occupation holders) and not the specific occupational roles and/or dispositions that inhere. Without the specific roles and dispositions, the design plan suggests, for instance, that "occupation role" would be a role of a pharmacist—and of every other occupation holder—rather than "pharmacist role." And there would be no disposition to which the abilities, skills, and intentions that go into being a pharmacist collectively contribute.

**Expected behavior**
For each occupation holder, there should be a corresponding occupation role (e.g., Occupation role > Welder role is a "role of" Welder) and corresponding occupation disposition (e.g., Occupation disposition > Welder disposition is a "disposition of" Welder).

**Screenshots**
![OccO Design Plan](https://github.com/Occupation-Ontology/OccO/assets/109489205/5af60f8e-cc33-443a-91ed-162854e9ae5e)

**Desktop (please complete the following information):**
 - Windows 10 Pro
 - Firefox
 - Version 117.0

**Comment by giacomodecolle** [comment]
Hi @mattolds, I absolutely agree with you that OccO needs to push more focus into the role and disposition parts of the hierarchy in order to reflect the paper. Notice that an entity is an occupation holder if they have either an occupation disposition or role. This means that we do not need to replicate the full occupation holder hierarchy in both places, but only in one depending on what we need. For example, "baker" can refer to each of three (holder, role, disposition), but it's likely that taxonomies like O NET and ESCO are referring to the job, i.e. the role, rather than the disposition (that would be more, "unemployed baker").

In any case, this means that first of all we need to provide two axioms that define the occupation holder correspondingly. One of these axioms was already added for occupation dispositions. @zhengj2007 would you please add one for occupation roles too?

Also, I think that most if not all of the current subclasses of occupation holder are best understood as referring to roles. Instead of replicating the hierarchy under role, we might want to move it to occupation role directly, then leave occupation holder as a defined class.

**Comment by mattolds** [comment]
Hi @giacomodecolle. That's a much neater solution. I was on Protege and didn't notice the new axiom. On the other hand, it seems like if you are an occupation holder, you must be bearing an occupation role. In that case, we could define "occupation holder" with one axiom.

If the hierarchy is moved under "occupation role," and "occupation holder" becomes a defined class, what about the occupation disposition hierarchy?

**Comment by giacomodecolle** [comment]
@mattolds you are right that this is likely not visible in Protege yet because @zhengj2007 updated the Robot template and perhaps hasn't run them yet to produce the .owl file.

Quoting the paper, "An occupation holder is someone bearing either an occupation role or an occupation disposition". This means that if you have either of the two, you are also an occupation holder. It might be that you just have an occupation disposition but no role, i.e. you are an unemployed programmer looking for a job, and then you still are an occupation holder.

The three hierarchies mirror each other, but most of the work would be done by the role classes. We can then add ad hoc terms under disposition if they are needed by the taxonomies we are trying to incorporate (ESCO, O NET, SOC, etc.)

**Comment by zhengj2007** [comment]
I added: 
occupation holder = 'Homo sapiens' and 'has disposition' some 'occupation disposition'

I did not added the role axiom based on my understanding of OccO discussion. Apologized that I have not read the OccO paper yet.

I think all 'occupation role' needs corresponding 'occupation disposition'. But not all 'occupation disposition' need the corresponding 'occupation role'. That's why I only add 'occupation disposition' as equivalent. I can add 'occupation role' as subClass axiom.

Based on the OBO operation committee discussion, it is bad ontology development practice to mirror the hierarchy under different entity, such as for 'teacher', we define 'teacher occupation', 'teacher disposition' and 'teacher role' in the ontology. My view is we may need to mirror the hierarchy for some applications. But we will add those only when we have real application. 

We can discuss the issue on the OccO call.

**Comment by giacomodecolle** [comment]
I agree with you that we don't need to mirror the full hierarchy unless there are specific uses for us to do so. This is exactly the reason why we should add an axiom for role and one for disposition, in case that we end up deciding that the whole role hierarchy doesn't need to be mirrored under disposition or viceversa.
